tenets/codelingo/go/pass-struct-by-pointer/example.go

```diff
@@ -0,0 +1,44 @@
+package main
+
+import (
+	"fmt"
+)
+
+type Cat struct {
+	Name string
+	age int
+}
+
+type dog struct {
+	name string
+	age int
+}
+
+func getCatName(c *Cat) {
+	fmt.Println(c.Name) // Issue, accessing struct with exported field(s) by reference
```

Comment:
hmm, I think this needs to be more sophisticated

I think the big issue is in *keeping* a reference, such that more than one piece of code could operate on it at the same time

Reply:
to elaborate: in these cases, the calling context can be safely assumed not to operate on it while the function is running, because it's not doing anything until the function returns

Reply:
…and, I guess, it's not so much about "exported" as it is "accessible" – assuming the code compiles (i.e. the fields are accessible) the problematic cases are still possible